core/analysis/ngram_token_stream.cpp

```diff
@@ -20,138 +20,206 @@
 /// @author Andrey Abramov
 ////////////////////////////////////////////////////////////////////////////////
 
-#include <rapidjson/rapidjson/document.h> // for rapidjson::Document
-#include <rapidjson/rapidjson/writer.h> // for rapidjson::Writer
-#include <rapidjson/rapidjson/stringbuffer.h> // for rapidjson::StringBuffer
+#include "ngram_token_stream.hpp"
 
 #include <frozen/unordered_map.h>
 #include <frozen/string.h>
 
-#include "ngram_token_stream.hpp"
+#include "velocypack/Slice.h"
+#include "velocypack/Builder.h"
+#include "velocypack/Parser.h"
+#include "velocypack/velocypack-aliases.h"
 #include "utils/hash_utils.hpp"
-#include "utils/json_utils.hpp"
+#include "utils/vpack_utils.hpp"
 #include "utils/utf8_utils.hpp"
 
 namespace {
 
-constexpr irs::string_ref MIN_PARAM_NAME               = "min";
-constexpr irs::string_ref MAX_PARAM_NAME               = "max";
-constexpr irs::string_ref PRESERVE_ORIGINAL_PARAM_NAME = "preserveOriginal";
-constexpr irs::string_ref STREAM_TYPE_PARAM_NAME       = "streamType";
-constexpr irs::string_ref START_MARKER_PARAM_NAME      = "startMarker";
-constexpr irs::string_ref END_MARKER_PARAM_NAME        = "endMarker";
+constexpr VPackStringRef MIN_PARAM_NAME               = VPackStringRef("min");
+constexpr VPackStringRef MAX_PARAM_NAME               = VPackStringRef("max");
+constexpr VPackStringRef PRESERVE_ORIGINAL_PARAM_NAME = VPackStringRef("preserveOriginal");
+constexpr VPackStringRef STREAM_TYPE_PARAM_NAME       = VPackStringRef("streamType");
+constexpr VPackStringRef START_MARKER_PARAM_NAME      = VPackStringRef("startMarker");
+constexpr VPackStringRef END_MARKER_PARAM_NAME        = VPackStringRef("endMarker");
 
 constexpr frozen::unordered_map<irs::string_ref, irs::analysis::ngram_token_stream_base::InputType, 2> STREAM_TYPE_CONVERT_MAP = {
```

Comment:
Why not use velocypack::StringRef here as keys? Anyway we are converting to StringRef while using.

Reply:
postponed